contracts/samples/SimpleAccount.sol

```diff
@@ -87,6 +80,10 @@ contract SimpleAccount is BaseAccount, UUPSUpgradeable, Initializable {
      */
     function initialize(address anOwner) public virtual initializer {
         _initialize(anOwner);
+        //increment nonce during construction.
+        // This way, the cost of moving from "0" to "1" is absorbed
+        // by first (construction) transaction, instead of during the 2nd transaction
+        entryPoint().incrementNonce(0);
```

Comment:
It was done so that even wallets that are not created via a dedicated UserOp, will have their next operation gas "clean".
A wallet can be created by execution its initCode either by the UserOp, or by other means. 
Calling "incrementNonce" the first time will cost 20000 gas, and any further call 5000 (I ignore for now explicitly setting to a new key, which is will cost 20000 again, but that an explicit operation a wallet owner does)
Not incrementing it during construction will mean first TX after construction will cost 20000 gas more than any further TX. We prefer to make gas-per-transaction as consistent as possible, and thus make only the construction transaction different.

Calling "incrementNonce" from the factory, indeed causes double-increment of the nonce, but the overhead is minimal (~400 gas)
The side-effect is indeed an extra increment of the nonce.